### PR TITLE
Integrate WebDev and build_runner daemon

### PR DESCRIPTION
**Pull request by grouma**
- Create new command `serve2` which will eventually replace `serve`
- The new command will:
  - be hidden for the time being
  - connect to the build daemon
  - create a proxy server for each requested build target 
    - this server forward requests to the daemon asset server
    - in the future this server will have hooks for hot reload / hot restart
- Add and E2E integration test
  - This test is expected to fail until `package:build_runner` version `1.2.1` is released

### CODE REVIEW

**Comment by grouma** [comment]
It's in the PR: https://github.com/dart-lang/webdev/pull/101/files#diff-69475c4febc8a4bd93ece8bbf53cd17fR63

**Comment by kevmoo** [comment]
Ah! It's a completely new command. Gotcha

File: webdev/lib/src/command/serve2_command.dart [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+  Serve2Command() : super(releaseDefault: false, outputDefault: outputNone) {
+    argParser
+      ..addOption('hostname',
+          help: 'Specify the hostname to serve on', defaultsTo: 'localhost')
+      ..addFlag('log-requests',
+          defaultsTo: false,
+          negatable: false,
+          help: 'Enables logging for each request to the server.')
+      // For development purposes.
```
Comment (natebosch) [inline webdev/lib/src/command/serve2_command.dart]: We talked offline about removing this

File: webdev/lib/src/command/serve2_command.dart [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    }
+
+    var manager = ServerManager(
+      getDaemonPort(workingDirectory),
+      hostname,
+      targetPorts,
+      logRequests,
+    );
+
+    print('Starting resource servers...');
```
Comment (natebosch) [inline webdev/lib/src/command/serve2_command.dart]: should we be going through a logger?

File: webdev/lib/src/serve/daemon_client.dart [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+/// Connects to the `build_runner` daemon.
+Future<BuildDaemonClient> connectClient(
+        String workingDirectory, List<String> options) =>
+    BuildDaemonClient.connect(
+      workingDirectory,
+      ['pub', 'run', 'build_runner', 'daemon']..addAll(options),
+    );
+
+/// Returns the port of the daemon asset server.
+int getDaemonPort(String workingDirectory) {
```
Comment (natebosch) [inline webdev/lib/src/serve/daemon_client.dart]: https://www.dartlang.org/guides/language/effective-dart/design#avoid-starting-a-method-name-with-get

`daemonPort` or `daemonPortForDirectory` would be fine.

File: webdev/lib/src/serve/handlers/asset_handler.dart [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+  Handler _handler;
+
+  AssetHandler(
+    this._daemonPort,
+    this._target,
+  );
+
+  Handler get handler {
+    _handler ??= Cascade()
```
Comment (natebosch) [inline webdev/lib/src/serve/handlers/asset_handler.dart]: Why do we need a `Cascade`?

```dart
Handler get handler => _handler ??= proxyHandler(...);
```

File: webdev/lib/src/command/serve2_command.dart (hunk 2 above)
Comment (grouma) [reply webdev/lib/src/command/serve2_command.dart]: In the future we can and probably should. All other webdev commands simply write to stdout.

**Pull request merged**